### PR TITLE
add use_populator property to PVC

### PR DESCRIPTION
**Pull request by dalia-frank**
##### Short description:
Add a function to determine if DV was deployed using populators or not
##### More details:
The function check the relevant annotation on the DV

### CODE REVIEW

File: ocp_resources/datavolume.py [hunk 1]
```diff
@@ -242,6 +242,14 @@ def wait(self, timeout=TIMEOUT_10MINUTES, failure_timeout=TIMEOUT_2MINUTES):
             status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
         )
 
+    def is_deployed_by_populators(self):
```
Comment (myakove) [inline ocp_resources/datavolume.py:245]: Set this function as `@property`
Rename to `deployed_by_populators`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/datavolume.py (hunk 1 above)
Comment (dalia-frank) [reply ocp_resources/datavolume.py]: ack

File: ocp_resources/datavolume.py [hunk 2]
```diff
@@ -252,13 +260,9 @@ def pvc(self):
...
     def scratch_pvc(self):
-        pvc_metadata = self.pvc.instance.metadata
         scratch_pvc_prefix = (
-            f"prime-{pvc_metadata.uid}"
-            if pvc_metadata.annotations.get(
-                f"{self.ApiGroup.CDI_KUBEVIRT_IO}/storage.usePopulator"
-            )
-            == "true"
+            f"prime-{self.pvc.instance.metadata.uid}"
+            if self.is_deployed_by_populators
```
[the remark this answers is not in the document]
Comment (dalia-frank) [reply ocp_resources/datavolume.py:266]: ack

[the remark this answers is not in the document]
Comment (dalia-frank) [reply ocp_resources/datavolume.py:266]: you are right :) thanks

File: ocp_resources/datavolume.py [hunk 3]
```diff
@@ -252,13 +261,9 @@ def pvc(self):
...
```
Comment (myakove) [inline ocp_resources/datavolume.py:266]: no such function `self.is_deployed_by_populators`, you just rename it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/datavolume.py [hunk 4]
```diff
@@ -242,6 +242,15 @@ def wait(self, timeout=TIMEOUT_10MINUTES, failure_timeout=TIMEOUT_2MINUTES):
             status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
         )
 
+    @property
+    def deployed_by_populators(self):
```
Comment (rnetser) [inline ocp_resources/datavolume.py:246]: this is a pvc property, not dv

Comment (rnetser) [reply ocp_resources/datavolume.py:246]: also please rename to match what is checked `usePopulator`

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Review by jpeimer — APPROVED** [review]

**Pull request merged**